server/internal/store/file.go

```diff
@@ -0,0 +1,148 @@
+package store
+
+import (
+	"time"
+
+	"gorm.io/gorm"
+)
+
+// FileStatus represents the status of a file.
+type FileStatus string
+
+// LastErrorCode represents the error code of the last error.
+type LastErrorCode string
+
+// ChunkingStrategyType represents the type of chunking strategy.
+type ChunkingStrategyType string
+
+const (
+	// FileStatusInProgress represents the in_progress status.
+	FileStatusInProgress FileStatus = "in_progress"
+	// FileStatusCompleted represents the completed status.
+	FileStatusCompleted FileStatus = "completed"
+	// FileStatusFailed represents the failed status.
+	FileStatusFailed FileStatus = "failed"
+	// FileStatusCancelled represents the cancelled status.
+	FileStatusCancelled FileStatus = "cancelled"
+
+	// LastErrorCodeNone represents no error.
+	LastErrorCodeNone LastErrorCode = ""
+	// LastErrorCodeServerError represents a server error.
+	LastErrorCodeServerError LastErrorCode = "server_error"
+	// LastErrorCodeRateLimitExceeded represents a rate limit exceeded error.
+	LastErrorCodeRateLimitExceeded LastErrorCode = "rate_limit_exceeded"
+
+	// ChunkingStrategyTypeAuto represents the auto chunking strategy.
+	ChunkingStrategyTypeAuto ChunkingStrategyType = "auto"
+	// ChunkingStrategyTypeStatic represents the static chunking strategy.
+	ChunkingStrategyTypeStatic ChunkingStrategyType = "static"
+	// ChunkingStrategyTypeOther represents the other chunking strategy.
+	ChunkingStrategyTypeOther ChunkingStrategyType = "other"
```

Comment:
Do we need `other` for the chunking strategy as it is not mentioned in the API reference.